### PR TITLE
Creating tests for bgg companion api

### PR DESCRIPTION
**Pull request by JDGiardino**
Addresses the story - https://github.com/JDGiardino/BGG-Companion/issues/31

This PR is creating testing for bgg_compnaion_api.py using pytest (will attach a screenshot once PR is complete showing all tests are passing).  Here are some of the changes made in order to write tests for this module : 

- A class was created in bgg_compnaion_api.py for all methods so that when the class is instantiated, functions can either use the normal RequestsRetryClient **_OR_** a mock request client designed for testing
- Created a data class within tests called TestBggCompanionVariables.py which houses all the mock data and test variables used by the main test_bgg_companion.py.  **_NOTE_** The data in this TestBggCompanionVariables.py class is very long but just contains sample test data taken from API responses.
- Updated some variables in data class to be optional
- Creating a Github Action that runs pytest using documentation https://github.com/snok/install-poetry and https://www.techiediaries.com/python-unit-tests-github-actions/

### CODE REVIEW

**Comment by JDGiardino** [comment]
I just realized how I broke the Github workflow.  It might be helpful to have some yaml file validation so I don't make a mistake like that again

File: src/bgg_companion_api.py [hunk 1]
```diff
@@ -12,91 +12,103 @@
 from cachetools import cached, TTLCache
 
 
-def request(url: str):
-    response = RequestsRetryClient().request(method="GET", url=url)
-    return response.text
+class BggCompanionApi(object):
```
Comment (nickdibari) [inline src/bgg_companion_api.py:15]: Just a reminder: did you update the `app.py` file to work with the new format for this module? Or is that abstracted away somehow?

File: .github/workflows/lint.yml [hunk 2]
```diff
@@ -37,3 +37,4 @@ jobs:
         #uses: wearerequired/lint-action@v2.0.0
         #with:
           #black : true
+
```
[the remark this answers is not in the document]
Comment (JDGiardino) [reply .github/workflows/lint.yml]: Awesome!!!  This seems to have worked.  Thanks for the suggestion

File: src/bgg_companion_api.py (hunk 1 above)
Comment (JDGiardino) [reply src/bgg_companion_api.py:15]: Ahhhh yes yes, this was one of the many changes I had locally all messed up and not committed to the branch

**Review by ndibari-etsy — APPROVED** [review]

File: tests/models/TestBggCompanionVariables.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
+from dataclasses import dataclass
+from collections import OrderedDict
+
+from src.models.BoardGame import BoardGame
+
+
+class TestGetCollectionVariables:
```
Comment (halo123u) [inline tests/models/TestBggCompanionVariables.py:7]: Maybe rename this to `TestGetCollectionVariables` to `TestGetCollectionData`

Comment (JDGiardino) [reply tests/models/TestBggCompanionVariables.py]: Definitely makes sense, will do!

**Pull request merged**